### ISSUE TITLE
Max number of createSelector arguments

### ISSUE THREAD

**Issue by UserGalileo** [ngrx/platform#1192]
There is currently a limit of 8 selectors which can be composed in a single createSelector call. I’d like to understand the choice, and if it’s a matter of performance, I think everyone should be able to decide to compose more than 8 selectors knowing that performances could decrease. I am on a large enterprise app which contains hundreds of entities and I need to create selectors which compose up to 15/20 entities.

## Describe any alternatives/workarounds you're currently using
I am currently splitting my selectors in 2/3 selectors and then combining them again. Which feels pretty useless and worse in terms of performance since it adds 1 or 2 functions to the list.

## Other information:
None. If you have alternatives or suggestions, I’ll be glad to hear them :)

## If accepted, I would be willing to submit a PR for this feature

[ ] Yes (Assistance is provided if you need help submitting a pull request)
[x ] No

**Comment by timdeschryver** [comment]
Should we increase the number of `createSelector` types, if so how many should we support?

**Comment by UserGalileo** [comment]
@brandonroberts That's exactly what I needed, I had found it in the source code bus I was unsure if it was the right tool, thanks. Could this be written explicitly in the documentation? I'm unable to find it anywhere (like, it appears nobody is talking about it in the entire web) :)

**Comment by UserGalileo** [comment]
@brandonroberts Yup, I'm writing things down :) Just wanted to ask one thing: as of now, in order to change the memoization process, we need to pass a different `MemoizeFn` to `createSelectorFactory`, but I feel (I'm still thinking, I may be wrong) that it's not exactly the memoization process that we may need to change, but just the comparers.
Correct me if I'm wrong, but as of now, for example, `defaultMemoize`, in its memoize() function uses the keyword `arguments`. If I had to write another `MemoizeFn` I should know who calls memoize() from the outside and which parameters it passed through, I should know how to compose that function, etc...
Isn't it a bit strange? I would expect, if I had to rewrite the memoization logic, to just interfere with the arguments and the result. Like, somewhere over these steps:

```
if (!isArgumentsChanged(arguments, lastArguments, isArgumentsEqual)) {
   return lastResult;
}

const newResult = projectionFn.apply(null, arguments);
if (isResultEqual(lastResult, newResult)) {
  return lastResult;
}
```

To be given just the values, and return a boolean for the arguments and one for the result. What do you think? (Again, tell me if I'm missing something, I got into the source code like 2 hours ago)
